File: occasum-core/src/main/java/pw/stamina/occasum/node/SimplePropertyNode.java
package pw.stamina.occasum.node;

import pw.stamina.occasum.PropertyHandle;
import pw.stamina.occasum.node.factory.PropertyNodeFactory;
import pw.stamina.occasum.properties.Property;

import java.util.Objects;

final class SimplePropertyNode extends AbstractParentedPropertyNode {
    private final Property property;

    SimplePropertyNode(PropertyNodeFactory factory,
                       PropertyHandle handle,
                       PropertyNode parent,
                       Property property) {
        super(factory, handle, parent);
        Objects.requireNonNull(property, "property");
        this.property = property;
    }

    @Override
    public String getName() {
        return property.getName();
    }

    @Override
    public String getId() {
        return property.getId();
    }

    @Override
    public Property getProperty() {
        return property;
    }

    @Override
    public boolean hasProperty() {
        return true;
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/node/SimplePropertyNode.kt
package pw.stamina.occasum.node

import pw.stamina.occasum.PropertyHandle
import pw.stamina.occasum.node.factory.PropertyNodeFactory
import pw.stamina.occasum.properties.Property

internal class SimplePropertyNode(
        factory: PropertyNodeFactory,
        handle: PropertyHandle,
        parent: PropertyNode,
        override val property: Property<*>
) : AbstractParentedPropertyNode(factory, handle, parent) {

    override val name: String = property.name

    override val id: String = property.id
}
